ray-operator/controllers/ray/raycluster_controller_fake_test.go

```diff
@@ -365,6 +373,158 @@ func tearDown(t *testing.T) {
+
+	testPodName := "eventPod"
+
+	// testPodEventName is the name of the event that will be created for testPodName
+	// The name of the event is generated by concatenating the pod name and a
+	// meaningless random string
+	testPodEventName := fmt.Sprintf("%s.15f0c0c5c5c5c5c5", testPodName)
+
+	// add a pod in a different namespace
+	newPods := []runtime.Object{
+		&corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testPodName,
+				UID:       types.UID(testPodName),
+				Namespace: "ns2",
+				Labels: map[string]string{
+					common.RayNodeLabelKey:      "yes",
+					common.RayClusterLabelKey:   instanceName,
+					common.RayNodeGroupLabelKey: groupNameStr,
+				},
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:    "ray-worker",
+						Image:   "rayproject/ray:2.2.0",
+						Command: []string{"echo"},
+						Args:    []string{"Hello Ray"},
+					},
+				},
+			},
+			Status: corev1.PodStatus{
+				Phase: corev1.PodRunning,
+			},
+		},
+		&corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "newPod2",
+				UID:       types.UID("newPod2"),
+				Namespace: namespaceStr,
+				Labels: map[string]string{
+					common.RayClusterLabelKey:   instanceName,
+					common.RayNodeGroupLabelKey: groupNameStr,
+				},
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:    "ray-worker",
+						Image:   "rayproject/ray:2.2.0",
+						Command: []string{"echo"},
+						Args:    []string{"Hello Ray"},
+					},
+				},
+			},
+			Status: corev1.PodStatus{
+				Phase: corev1.PodRunning,
+			},
+		},
+		&corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testPodName,
+				UID:       types.UID(testPodName),
+				Namespace: namespaceStr,
+				Labels: map[string]string{
+					common.RayNodeLabelKey:      "yes",
+					common.RayClusterLabelKey:   instanceName,
+					common.RayNodeGroupLabelKey: groupNameStr,
+				},
+				Annotations: map[string]string{
+					common.RayFTEnabledAnnotationKey: "true",
+				},
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:    "ray-worker",
+						Image:   "rayproject/ray:2.2.0",
+						Command: []string{"echo"},
+						Args:    []string{"Hello Ray"},
+					},
+				},
+			},
+			Status: corev1.PodStatus{
+				Phase: corev1.PodRunning,
+			},
+		},
+	}
+
+	testPods = append(testPods, newPods...)
+
+	fakeClient := clientFake.NewClientBuilder().WithRuntimeObjects(testPods...).Build()
+
+	podList := corev1.PodList{}
+	err := fakeClient.List(context.Background(), &podList, client.InNamespace(namespaceStr))
+
+	assert.Nil(t, err, "Fail to get pod list")
+	assert.Equal(t, len(testPods)-1, len(podList.Items), "Init pod list len is wrong")
+
+	testRayClusterReconciler := &RayClusterReconciler{
+		Client:   fakeClient,
+		Recorder: &record.FakeRecorder{},
+		Scheme:   scheme.Scheme,
+		Log:      ctrl.Log.WithName("controllers").WithName("RayCluster"),
+	}
+
+	// add event for reconcile
```

Comment:
Does it make sense to also add an event which should get filtered out and not trigger a reconcile?